content(blog/events): add trip report of 2026 London summit (#8840)
nodejs/nodejs.org: apps/site/pages/en/blog/events/collab-summit-2026-london.md

---
date: '2026-04-24T00:00:00.000Z'
category: events
title: 'Trip report: Node.js collaboration summit (2026 London)'
layout: blog-post
author: Chengzhong Wu, Joyee Cheung
---

This April, the first [Node.js Collaboration Summit](https://nodejs.org/en/about/get-involved/collab-summit) of 2026 was hosted by [Bloomberg][] in London. In this edition, we welcomed over 40 in-person participants, and around a dozen more joined remotely.

Here is a recap of what happened at the summit.

### Next 10

> In this session, [Jacob Smith][] started an on-site review of the questions that would be asked in the [Next-10 survey for 2026](https://github.com/nodejs/next-10/issues/369), including suggestions to existing questions, questions and options to add or remove, and recent AI discussions.

Reviewer: ```suggestion
In this session, [Jacob Smith][] reviewed results from the Collaborator Health survey and compared health numbers between 2025 and 2026, as well as some key highlights from the 2025 user survey (including new areas, which have generally different usage); he and [Marco Ippolito][]  led an on-site review of the questions that would be asked in the [Next-10 survey for 2026](https://github.com/nodejs/next-10/issues/369), including suggestions to existing questions, questions and options to add or remove, and recent AI discussions.
```

### New Release Schedule

Rafael Gonzaga announced a [new release schedule](https://nodejs.org/en/blog/announcements/evolving-the-nodejs-release-schedule) going into effect starting with Node.js v27. TL;DR, starting with Node.js v27, the Node.js version numbers align with the calendar year of their initial Current release!

This is a reflection of the current Node.js volunteer-based maintenance and an effort to keep the Node.js project sustainable in the long run. When it comes to security vulnerabilities, managing security releases across four or five active release lines has become difficult to sustain in the current Node.js voluntary work model. By reducing the number of concurrent release lines, the project can focus on better supporting the releases people actually use.

> What's important is what's not changed with the new release schedule:

Reviewer 2: ```suggestion
What's important is what's not changed with the new release schedule:

```
I think we just need to add a new line here for linting/prettier to pass

### New Streams API

[James Snell][] presented a new, more unified Streams API for the Web and Node.js. Historically, Node.js has relied on the highly optimized `node:stream`. In recent years, Node.js has also implemented WHATWG Web Streams to ensure cross-platform compatibility with browsers, Deno, Bun, and edge environments (like Cloudflare Workers). However, managing two different stream ecosystems has created friction for developers.

The Web Streams API has significant promise and allocation overhead. It was designed primarily for browsers, and when applied to high-throughput backend environments, the architectural cost becomes obvious.

The proposed New Streams API takes advantage of modern JavaScript Async Iteration, treats streams natively as async iterables, leaning heavily into `async/await` and `for await...of` loops, which is how modern developers naturally write code. Handling a filled data buffer has historically been a confusing black box. The new API also proposes forcing developers to explicitly choose a backpressure strategy when a buffer is full.

> 

Reviewer 3: There is a very long part of back-and-forth in the recording, I tried to capture some points..

```suggestion

At the summit, we collected feedback on the design, compatibility with existing stream APIs, performance implications, additional use cases, whether the new proposal can help address some common issues in the existing APIs, and discussed whether the new Streams API can be implemented on the Web.
```

### Node.js Collaboratorship

[Jacob Smith][] led the session on exploring ways to lower the barrier to entry for developers who want to contribute to the Node.js core, ensuring the project remains accessible to new talent.

> Triaging issues and reviewing pull requests takes a massive amount of maintainer bandwidth. There were also discussions about enforcing code ownership to improve review cycles. However, in the Node.js voluntary work model, code ownership also comes with difficulties. We could also decouple collaborators from the commit bits, encourage non-collaborators to review, and build trust. This could be a lower-barrier pathway for people to contribute.

Reviewer 3: I think we should try to frame the report to reflect what was discussed without showing inclination on whether the proposal is preferrable or will go through (from what I can gather at the summit, most sessions are just food for thought and we didn't decide whether we'd actually act on it - we'd only know after it's back on GitHub).

```suggestion
Triaging issues and reviewing pull requests takes a massive amount of maintainer bandwidth. At the session we discussed ideas to enforce code ownership to encourage collaborators to take over more review work. Since in the Node.js voluntary work model, this can come with difficulties, we also brainstormed ideas about decoupling collaboratorship from the commit bits and extending reviewership to working group/team members, in order to encourage non-collaborators to review and build trust.
```

> Triaging issues and reviewing pull requests takes a massive amount of maintainer bandwidth. At the session we discussed ideas to enforce code ownership to encourage collaborators to take over more review work. Since in the Node.js voluntary work model, this can come with difficulties, we also brainstormed ideas about decoupling collaboratorship from the commit bits and extending reviewership to working group/team members, in order to encourage non-collaborators to review and build trust.

Reviewer: > do we have consensus in the room about this and this? no objections? then at least we have consensus in the room!

This did happen. Crap—was it when you had to duck out coughing? 😨

> My impression is that we only collected a lot of feedback and the room gravitated towards something roughly in a direction and will continue iterating on this in GitHub

There was definitely a lotta good feedback. From speaking to others afterwards, they also seemed to understand we had ultimately reached a fairly concrete consensus and path forward though.

> Personally I remember I raised the point that core subsystems are the wrong way to group people and we discussed at length about fs as an example.

I recall the FS point (it was a good one), but my recollection was that that was resolved (specifically, James had the winning idea). If it's not, could you please let me know what part of the resolution was incomplete?

> but specifically how the teams should be formed is very much in the air

Could you clarify this? I don't recall ambiguity on this in the end.

PS I wrote up [a summary](https://github.com/openjs-foundation/summit/issues/486#issuecomment-4319754806) and tagged you on it. I think there would be a better place to have this discussion.

> Triaging issues and reviewing pull requests takes a massive amount of maintainer bandwidth. At the session we discussed ideas to enforce code ownership to encourage collaborators to take over more review work. Since in the Node.js voluntary work model, this can come with difficulties, we also brainstormed ideas about decoupling collaboratorship from the commit bits and extending reviewership to working group/team members, in order to encourage non-collaborators to review and build trust.

Reviewer 3: Yeah I think that might've happened when I stepped out. I'll take a look at the summary when I have the cycles, thanks. For the trip report specifically I think we should avoid talking about concrete consensus if there's any doubt and just reflect what's been discussed.

> Triaging issues and reviewing pull requests takes a massive amount of maintainer bandwidth. At the session we discussed ideas to enforce code ownership to encourage collaborators to take over more review work. Since in the Node.js voluntary work model, this can come with difficulties, we also brainstormed ideas about decoupling collaboratorship from the commit bits and extending reviewership to working group/team members, in order to encourage non-collaborators to review and build trust.

Reviewer: Okay 🙂 as you said, it would be limited to the room anyway, so not a full consensus—nothing landed yet.

```suggestion
We reached broad-strokes of a plan forward via a new model, applying "code owners" to Core subsystems via teams wherein team members can approve (and block) PRs to their subsystem; team members would now be collaborators (with the new, narrowed purview and lower barrier to entry). A new "Maintainer" tier of membership would also be created with organisation-wide authority (similar to the current "collaborator" role) with a moderately high barrier to entry.
```

> Triaging issues and reviewing pull requests takes a massive amount of maintainer bandwidth. At the session we discussed ideas to enforce code ownership to encourage collaborators to take over more review work. Since in the Node.js voluntary work model, this can come with difficulties, we also brainstormed ideas about decoupling collaboratorship from the commit bits and extending reviewership to working group/team members, in order to encourage non-collaborators to review and build trust.

Author: This is not a document capturing consensus or conclusions. I would prefer only describing "what's been discussed at the summit", rather than "what's been concluded at the summit" or "what plan has been reached".

The consensus building should go through the usual procedure, like for this particular topic, opening a dedicated issue in `nodejs/admin`.

### Node.js use of AI contributions

This session, led by [Jacob Smith][], was kicked off by the gigantic VFS pull request (though over half of it consisted of integration tests), which openly indicated that an AI-assisted coding tool had been used to compose the PR.

While AI makes the cost of generating code practically zero, the cost of reviewing it remains strictly linear and bound by human limits. Thoroughly reviewing a 19,000-line PR requires weeks or even months of dedicated human effort. Pushing massive AI generations onto reviewers was described by some attendees as a "denial of service" on maintainer bandwidth.

An outright ban on AI could be infeasible for the Node.js project to implement. Mandating the disclosure of AI tools might inadvertently force contributors to publicize their disabilities or medical needs, which challenges project inclusiveness and raises legal and ethical concerns.

> Rather than defaulting to an outright ban on AI, the consensus leaned toward adapting open-source governance for the modern era. Strategies discussed include enforcing a process that requires lightweight design documents, RFCs, and explicit maintainer buy-in before a large PR (regardless of whether it is human-authored or not), and requiring that the human contributor submitting the code assumes full responsibility for its quality, security, and integration.

Reviewer 3: This is a controversial topic and to me it seemed we didn't reach any consensus at the summit. We should try avoid drawing any conclusions in the trip report lest readers think any decisions are made.

```suggestion
A few weeks before the summit, a large pull request implementing VFS in Node.js with the assistance of an AI coding agent led to debates over the use of AI in Node.js core. In this session, [Jacob Smith][] walked the audience through the background of this controversy, the legal opinion from the OpenJS foundation, and the current status of the use of AI in other OSS projects.

We then started listing pros & cons, concerns and thoughts about the use of AI in Node.js core using a retrospective board. There were a lot of diverse opinions about this topic. At the session, we discussed the challenges from reviewer bandwidth exacerbated by the use of AI, the ethical concerns about AI, whether a ban is feasible or desirable, whether a disclosure on the use of AI may help or draw further legal and ethical concerns, how AI may help maintenance and lower barrier of entry, and ideas on reducing the amount of noise caused by the use of AI.

We also brainstormed how to adapt open-source governance for the modern era. Strategies discussed include enforcing a process that requires lightweight design documents, RFCs, explicit maintainer buy-in before a large PR, and pledge from the human contributor submitting the code who should take full responsibility for its quality, security, and integration.
```

Joyee also presented [a new design for the vm Module APIs](https://github.com/nodejs/node/issues/62720) that have been experimental for 9 years in order to address known issues and finally bring it to stabilization. We collected feedback on how to integrate it with WebAssembly modules to ensure it's future-proof with ongoing ESM integration proposals in standardization bodies.

### Libuv v2

[Santiago Gimeno][] shared that after more than a decade on version 1, there is a renewed push to launch libuv [v2](https://github.com/libuv/libuv/issues/4622), which introduces necessary breaking changes to clean up the codebase, remove legacy APIs, and improve cross-platform consistency—capabilities already being leveraged by [Julia](https://julialang.org/).

> To address the ABI compatibility concern in Node.js, [Node-API](https://nodejs.org/api/n-api.html) can be a way out. However, [`napi_get_uv_event_loop`](https://nodejs.org/api/n-api.html#napi-get-uv-event-loop) exposes the libuv loop and is vulnerable to libuv ABI changes. Fortunately, this is limited and the only API being impacted. The documentation also warns about this potential impact.

Reviewer 3: It would be better to leave the wording a bit more inconclusive, since we don't make decisions at the summits.

```suggestion
As migrating to linuv v2 can break the ABI, we discussed ideas on how to mitigate it, for example by leveraging [Node-API](https://nodejs.org/api/n-api.html), and the nuances in this approach e.g. [`napi_get_uv_event_loop`](https://nodejs.org/api/n-api.html#napi-get-uv-event-loop) can still be vulnerable to libuv ABI changes, though its use is limited and it's ABI stability is warned in the documentation.
```

> As migrating to libuv v2 can break the ABI, we discussed ideas on how to mitigate it, for example by leveraging [Node-API](https://nodejs.org/api/n-api.html), and the nuances in this approach e.g. [`napi_get_uv_event_loop`](https://nodejs.org/api/n-api.html#napi-get-uv-event-loop) can still be impacted by libuv ABI changes, though its use is limited and its ABI stability is warned in the documentation.

Reviewer 3: ```suggestion
As migrating to libuv v2 can break the ABI, we discussed ideas on how to mitigate it, for example by leveraging [Node-API](https://nodejs.org/api/n-api.html), and the nuances in this approach e.g. [`napi_get_uv_event_loop`](https://nodejs.org/api/n-api.html#napi-get-uv-event-loop) can still be impacted by libuv ABI changes, though its use is limited and its ABI stability is warned in the documentation. We also discussed getting help to maintain v1 with security patches for a limited timespan, how to bring back `io_uring`, and which Node.js can start to ship libuv v2 (a very tentative timeline could be in 27).
```

### Node.js Virtual File System

[Matteo Collina][] presented the proposal for a Node.js built-in Virtual File System. By taking concepts previously explored in userland libraries (like `@platformatic/vfs`) and standardizing them into a core `node:vfs` module, Node.js can intercept standard filesystem calls and route them through a virtualized, memory-based layer. Developers can define specific data sources in memory (providers) and "mount" them so the runtime treats them exactly like local directories. The proposal also provides the ability to layer virtual filesystems on top of one another, or place a virtual layer directly over the physical disk to safely mock or override files.

> Userland VFS implementations require massive monkey-patching; moving it to core provides deep integration and supports more use cases like SEA.

Reviewer 3: I think this covered most points discussed:

```suggestion
In this session, Matteo walked the audience through the motivation of this feature: making file system virtualization easier for single-executable applications and testing, and replaces existing brittle practices that achieve this through user-land monkey-patching. Following a brief overview of the implementation architecture and the design choices, we collected more feedback on the PR, including stack trace visibility, observability, security, worker thread propagation, the ability to toggle it in specific paths, and support in native code. We'll continue iterating on this PR in GitHub.
```

> 

Reviewer 3: There are some discussions in this session that we should capture too.

### Node.js Security - State of the Ecosystem & What's Next

> [Rafael Gonzaga][] shared that the security team has recently advanced the ecosystem through a refined threat model, improved permission models, and enhanced release automation, but these efforts are currently being overshadowed by a massive influx of AI-generated vulnerability reports. This industry-wide surge, driven largely by users seeking CVE attribution and financial bounties, has severely strained maintainer capacity with high-noise, duplicative submissions that often lack reproduction steps or misclassify standard bugs as severe security threats. Despite attempted mitigations like pausing bug bounties, raising HackerOne signal requirements, and clarifying guidelines, the overwhelming volume has significantly driven up resolution times. To combat this bottleneck, the team is exploring strategies such as securing early access for proactive testing, attempting to alter reporting agent behaviors, and adopting a public security flow to bypass embargoes and speed up CI testing.

Reviewer 3: Looks like we overlooked the VEX effort.

```suggestion
[Rafael Gonzaga][] shared that the security team has recently advanced the ecosystem through a refined threat model, improved permission models, and enhanced release automation, and the new VEX (Vulnerability Exploitability eXchange) files that the security team is working on to reduce false positives for security scanners. 

These efforts are currently being overshadowed by a massive influx of AI-generated vulnerability reports. This industry-wide surge, driven largely by users seeking CVE attribution and financial bounties, has severely strained maintainer capacity with high-noise, duplicative submissions that often lack reproduction steps or misclassify standard bugs as severe security threats. Despite attempted mitigations like pausing bug bounties, raising HackerOne signal requirements, and clarifying guidelines, the overwhelming volume has significantly driven up resolution times. To combat this bottleneck, the team is exploring strategies such as securing early access for proactive testing, attempting to alter reporting agent behaviors, and adopting a public security flow to bypass embargoes and speed up CI testing.
```

> 

Reviewer 3: IIRC Robin presented a significant part of this. There was also the introduction of the two new programs which we should cover in the trip report.

Author: I could not find any public reference on the program so I didn't mention it here.

Reviewer 3: Oh actually I realized that part is in a separate session, though the recording is public https://www.youtube.com/watch?v=Vr2nrYENzSg

Reviewer 4: Please skip the new Security Stewardship Program in the written trip report until we formally announce this expanded partner initiative. We're still working through the potential for early access for partners.

> 

Reviewer 3: ```suggestion
In the session we revisited a thought-provoking proposal of making the security triage and bugfix process completely public to prevent a false sense of security in the age of AI, and to get rid of the CI embargo that has made testing inefficient. We discussed a few middle ground solutions, including keeping extending the visibility to more members in the organization, handling vulnerabilities of different severity with different visibility settings, and creating better documentations to drive AI agents away from generating false positive reports.
```

## Thanks

Thank you to all the attendees! Special appreciation goes to [Bloomberg][] for hosting the summit and creating a welcoming space for the Node.js community.

A big thanks as well to Thomas Chetwin ([@tchetwin](https://bsky.app/profile/tchetwin.bsky.social)), [Chengzhong Wu][], [Matteo Collina][], [Joyee Cheung][], and the OpenJS Foundation for organizing and making this event possible.

The Node.js Collaboration Summit recordings are now available at [YouTube](https://www.youtube.com/playlist?list=PLfMzBWSH11xZhA93H_9ulECtLVWtSm6zy).

[Jacob Smith]: https://github.com/JakobJingleheimer
[Rafael Gonzaga]: https://github.com/RafaelGSS
[Chengzhong Wu]: https://github.com/legendecas
[Joyee Cheung]: https://github.com/joyeecheung
[Matteo Collina]: https://github.com/mcollina
[Stephen Belanger]: https://github.com/Qard
[James Snell]: https://github.com/jasnell
[Santiago Gimeno]: https://github.com/santigimeno
[Bryan English]: https://github.com/bengl
[Marco Ippolito]: https://github.com/marco-ippolito

> [Bloomberg]: https://www.bloomberg.com/company/values/tech-at-bloomberg/

Reviewer: Alphabetise for grokkability and add Bruno Rodrigues

```suggestion
[Bloomberg]: https://www.bloomberg.com/company/values/tech-at-bloomberg/
[Bruno Rodrigues]: https://github.com/brunocroh
[Bryan English]: https://github.com/bengl
[Chengzhong Wu]: https://github.com/legendecas
[Jacob Smith]: https://github.com/JakobJingleheimer
[James Snell]: https://github.com/jasnell
[Joyee Cheung]: https://github.com/joyeecheung
[Marco Ippolito]: https://github.com/marco-ippolito
[Matteo Collina]: https://github.com/mcollina
[Rafael Gonzaga]: https://github.com/RafaelGSS
[Santiago Gimeno]: https://github.com/santigimeno
[Stephen Belanger]: https://github.com/Qard
```

---
date: '2026-04-24T00:00:00.000Z'
category: events
title: 'Trip report: Node.js collaboration summit (2026 London)'
layout: blog-post
author: Chengzhong Wu, Joyee Cheung
---

This April, the first [Node.js Collaboration Summit](https://nodejs.org/en/about/get-involved/collab-summit) of 2026 was hosted by [Bloomberg][] in London. In this edition, we welcomed over 40 in-person participants, and around a dozen more joined remotely.

> Here is a recap of what happened at the summit.

Reviewer 3: ```suggestion
The recording of the summit is available in [this playlist](https://www.youtube.com/playlist?list=PLfMzBWSH11xZhA93H_9ulECtLVWtSm6zy). Here is a recap of what happened at the summit.
```

### Next 10

In this session, [Jacob Smith][] reviewed results from the Collaborator Health survey and compared health numbers between 2025 and 2026, as well as some key highlights from the 2025 user survey (including new areas, which have generally different usage); he and [Marco Ippolito][] led an on-site review of the questions that would be asked in the [Next-10 survey for 2026](https://github.com/nodejs/next-10/issues/369), including suggestions to existing questions, questions and options to add or remove, and recent AI discussions.

### New Release Schedule

> Rafael Gonzaga announced a [new release schedule](https://nodejs.org/en/blog/announcements/evolving-the-nodejs-release-schedule) going into effect starting with Node.js v27. TL;DR, starting with Node.js v27, the Node.js version numbers align with the calendar year of their initial Current release!

Reviewer 3: ```suggestion
Rafael Gonzaga walked the audience through the new release schedule that will go into effect starting with Node.js v27. TL;DR, starting with Node.js v27, the Node.js version numbers align with the calendar year of their initial Current release!
```

IIUC there's no new announcement, it's just a walkthrough of something already announced before.

### New Streams API

[James Snell][] presented a new, more unified Streams API for the Web and Node.js. Historically, Node.js has relied on the highly optimized `node:stream`. In recent years, Node.js has also implemented WHATWG Web Streams to ensure cross-platform compatibility with browsers, Deno, Bun, and edge environments (like Cloudflare Workers). However, managing two different stream ecosystems has created friction for developers.

The Web Streams API has significant promise and allocation overhead. It was designed primarily for browsers, and when applied to high-throughput backend environments, the architectural cost becomes obvious.

The proposed New Streams API takes advantage of modern JavaScript Async Iteration, treats streams natively as async iterables, leaning heavily into `async/await` and `for await...of` loops, which is how modern developers naturally write code. Handling a filled data buffer has historically been a confusing black box. The new API also proposes forcing developers to explicitly choose a backpressure strategy when a buffer is full.

> At the summit, we collected feedback on the design, compatibility with existing stream APIs, performance implications, additional use cases, whether the new proposal can help address some common issues in the existing APIs, and discussed whether the new Streams API can be implemented on the Web.

Reviewer 3: ```suggestion
The new [`stream/iter`](https://nodejs.org/docs/latest-v25.x/api/stream_iter.html) API has landed in Node.js core and shipped in v25.9.0 as an experimental feature. At the summit, we collected feedback on the design, compatibility with existing stream APIs, performance implications, additional use cases, whether the new proposal can help address some common issues in the existing APIs, and discussed whether the new Streams API can be implemented on the Web.
```

### Node.js use of AI contributions

A few weeks before the summit, a large pull request implementing VFS in Node.js with the assistance of an AI coding agent led to debates over the use of AI in Node.js core. In this session, [Jacob Smith][] walked the audience through the background of this controversy, the legal opinion from the OpenJS foundation, and the current status of the use of AI in other OSS projects.

We then started listing pros & cons, concerns and thoughts about the use of AI in Node.js core using a retrospective board. There were a lot of diverse opinions about this topic. At the session, we discussed the challenges from reviewer bandwidth exacerbated by the use of AI, the ethical concerns about AI, whether a ban is feasible or desirable, whether a disclosure on the use of AI may help or draw further legal and ethical concerns, how AI may help maintenance and lower barrier of entry, and ideas on reducing the amount of noise caused by the use of AI.

> We also brainstormed how to adapt open-source governance for the modern era. Strategies discussed include enforcing a process that requires lightweight design documents, RFCs, explicit maintainer buy-in before a large PR, and pledge from the human contributor submitting the code who should take full responsibility for its quality, security, and integration.

Reviewer 3: ```suggestion
We also brainstormed how to adapt open-source governance for the modern era. Strategies discussed include enforcing a process that requires lightweight design documents, RFCs, explicit maintainer buy-in before a large PR, and pledge from the human contributor submitting the code who should take full responsibility for its quality, security, and integration. We'll continue the discussions and iteration of the policy documents on GitHub.
```
